src/mapping/cellranger_multi/script.py

```diff
@@ -140,19 +181,36 @@ def check_subset_dict_equal_length(group_name: str,
     assert len(set(lens.values())) <= 1, f"The number of values passed to {group_name} "\
                                          f"arguments must be 0, 1 or all the same. Offenders: {lens}"
 
-def process_params(par: dict[str, Any]) -> str:
-    # if par_input is a directory, look for fastq files
-    par["input"] = [Path(fastq) for fastq in par["input"]]
-    if len(par["input"]) == 1 and par["input"][0].is_dir():
-        logger.info("Detected '--input' as a directory, "
+def resolve_input_directories_to_fastq_paths(input_paths: list[str]) -> list[Path]:
+
+    input_paths = [Path(fastq) for fastq in input_paths]
+    if len(input_paths) == 1 and input_paths[0].is_dir():
+        logger.info("Detected a directory in input paths, "
                     "traversing to see if we can detect any FASTQ files.")
-        par["input"] = [input_path for input_path in par["input"][0].rglob('*')
+        input_paths = [input_path for input_path in input_paths[0].rglob('*')
                         if re.match(fastq_regex, input_path.name) ]
 
     # check input fastq files
-    for input_path in par["input"]:
+    for input_path in input_paths:
         assert re.match(fastq_regex, input_path.name) is not None, \
-               f"File name of --input '{input_path}' should match regex {fastq_regex}."
+            f"File name of --input '{input_path}' should match regex {fastq_regex}."
+
+    return input_paths
+
+def process_params(par: dict[str, Any]) -> str:
+
+    if par["input"]:
+        assert len(par["library_type"]) > 0, "--library_type must be defined when passing input to --input"
+        assert len(par["library_id"]) > 0, "--library_id must be defined when passing input to --input"
+
+        # if par["input"] is a directory, look for fastq files
+        par["input"] = resolve_input_directories_to_fastq_paths(par["input"])
+
+    # add helper input
+    helper_input = transform_helper_inputs(par)
+    for key in ["input", "library_id", "library_type"]:
+      par[key] = par.get(key, []) + helper_input[key]
```

Comment:
Sorry, my previous suggestion does not work because `par[key]` is always defined but can be

```suggestion
      par[key] = (par[key] if par[key] else []) + helper_input[key]
```